Outdated example to invoke Nim from C

The documentation of [Nim invocation example from C](https://nim-lang.org/docs/backends.html#interfacing-backend-code-calling-nim) is completely outdated.

### original issue

Nim compiler generates a statically linked library which does not link.

```gcc -o m -I./nimcache -Ipath/to/nim/lib libfib.a maths.c```
```
/tmp/cce625tz.o: In function `main':
maths.c:(.text+0x9): undefined reference to `NimMain'
maths.c:(.text+0x1c): undefined reference to `fib'
```

--- a/doc/backends.rst
+++ b/doc/backends.rst
@@ -268,7 +268,7 @@ form the Nim code, then link them into a static binary along your main C
 program::
 
   $ nim c --noMain --noLinking --header:fib.h fib.nim
-  $ gcc -o m -Inimcache -Ipath/to/nim/lib nimcache/*.c maths.c
+  $ gcc -o m -I$HOME/.cache/nim/fib_d -Ipath/to/nim/lib $HOME/.cache/nim/fib_d/*.c maths.c
 
 The first command runs the Nim compiler with three special options to avoid
 generating a ``main()`` function in the generated files, avoid linking the

